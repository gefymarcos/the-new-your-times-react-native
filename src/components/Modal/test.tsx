import React from 'react'
import { Provider } from 'react-redux'
import renderer, { act } from 'react-test-renderer'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'

import { fireEvent, render } from '@testing-library/react-native'

import BackDrop from '.'

jest.useFakeTimers()

const props = {
  title: 'Title of card',
  abstract: 'Description',
  url: 'https://myurl.com.br'
}

describe('<BackDrop />', () => {
  const middlewares = [thunk]
  const mockStore = configureStore(middlewares)

  it('should render correctly', () => {
    const { getAllByText } = render(
      <Provider store={mockStore({})}>
        <BackDrop {...props} />
      </Provider>
    )

    const title = getAllByText(/Title of card/i)
    const abstract = getAllByText(/Description/i)
    const url = getAllByText(/https:\/\/myurl.com.br/i)

    fireEvent(url[0], 'press')

    expect(title).toBeDefined()
    expect(abstract).toBeDefined()
    expect(url).toBeDefined()
  })

  it('should close modal', () => {
    const { getByTestId } = render(
      <Provider store={mockStore({})}>
        <BackDrop {...props} />
      </Provider>
    )

    const modal = getByTestId(/modal-custom/i)
    expect(modal.props.visible).toEqual(true)
    act(() => {
      modal.props.onRequestClose()
    })
    expect(modal.props.visible).toEqual(false)
    expect(modal).toBeDefined()
  })

  it('should close modal when press back button', () => {
    const { getByText, getByTestId } = render(
      <Provider store={mockStore({})}>
        <BackDrop {...props} />
      </Provider>
    )

    const button = getByText(/back/i)
    const modal = getByTestId(/modal-custom/i)
    expect(modal.props.visible).toEqual(true)

    fireEvent(button, 'press')

    expect(modal.props.visible).toEqual(false)
    expect(modal).toBeDefined()
  })

  it('should render correctly based on snapshot', () => {
    const tree = renderer
      .create(
        <Provider store={mockStore({})}>
          <BackDrop {...props} />
        </Provider>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
})
