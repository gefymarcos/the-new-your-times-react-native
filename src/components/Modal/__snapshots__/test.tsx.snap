// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BackDrop /> should render correctly based on snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
      "marginTop": 22,
    }
  }
>
  <Modal
    animationType="slide"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    testID="modal-custom"
    transparent={true}
    visible={true}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "marginTop": 22,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "white",
            "borderRadius": 20,
            "elevation": 5,
            "flex": 1,
            "marginTop": 50,
            "padding": 35,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 4,
            "width": "100%",
          }
        }
      >
        <View
          style={
            Object {
              "justifyContent": "center",
              "paddingVertical": 32,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#212121",
                "fontSize": 18,
                "fontWeight": "600",
                "marginHorizontal": 16,
                "marginVertical": 8,
              }
            }
          >
            Title of card
          </Text>
          <Text
            style={
              Object {
                "color": "#212121",
                "fontSize": 16,
                "lineHeight": 24,
                "marginHorizontal": 16,
                "marginVertical": 8,
              }
            }
          >
            Description
          </Text>
          <View
            accessibilityRole="button"
            accessibilityState={Object {}}
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {},
                ]
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#007AFF",
                      "fontSize": 18,
                      "margin": 8,
                      "textAlign": "center",
                    },
                  ]
                }
              >
                https://myurl.com.br
              </Text>
            </View>
          </View>
        </View>
        <View
          accessible={true}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "borderRadius": 20,
                "elevation": 2,
                "padding": 10,
              },
              Object {
                "backgroundColor": "#212121",
              },
            ]
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontWeight": "bold",
                "textAlign": "center",
                "width": 150,
              }
            }
          >
            Back
          </Text>
        </View>
      </View>
    </View>
  </Modal>
</View>
`
